crates/typed-store-macros/src/lib.rs

```diff
@@ -409,3 +410,217 @@ fn extract_generics_names(generics: &Generics) -> Vec<Ident> {
+///
+/// # Examples
+///
+/// Well formed struct of tables
+/// ```
+/// use rocksdb::Options;
+/// use typed_store::rocks::DBMap;
+/// use typed_store::Store;
+/// use typed_store_macros::StoreUtils;
+/// use typed_store::traits::StoreTableUtil;
+///
+/// /// Define a struct with all members having type Store<K, V>
+///
+/// fn custom_fn_name1() -> Options {Options::default()}
+/// fn custom_fn_name2() -> Options {
+///     let mut op = custom_fn_name1();
+///     op.set_write_buffer_size(123456);
+///     op
+/// }
+/// #[derive(StoreUtils)]
+/// struct Tables {
+///     /// Specify custom options function `custom_fn_name1`
+///     #[default_options_override_fn = "custom_fn_name1"]
+///     table1: Store<String, String>,
+///     #[default_options_override_fn = "custom_fn_name2"]
+///     table2: Store<i32, String>,
+///     // Nothing specifed so `typed_store::rocks::default_rocksdb_options` is used
+///     table3: Store<i32, String>,
+///     #[default_options_override_fn = "custom_fn_name1"]
+///     table4: Store<i32, String>,
+/// }
+///
+/// /// All traits in `DBMapTableUtil` are automatically derived
+/// /// Use the struct like normal
+/// #[tokio::main]
+/// async fn main() {
+/// let primary_path = tempfile::tempdir().expect("Failed to open temporary directory").into_path();
+/// /// This is auto derived
+/// let tbls_primary = Tables::open_tables_read_write(primary_path.clone(), None, None);
+///
+/// // Do some stuff with the DB
+/// }
+///
+/// // Bad usage example
+/// // Structs fields most only be of type Store<K, V>
+/// // This will fail to compile with error `All struct members must be of type DMBap<K, V>`
+/// // #[derive(DBMapUtils)]
+/// // struct BadTables {
+/// //     table1: Store<String, String>,
+/// //     bad_field: u32,
+/// // #}
+/// ```
+#[proc_macro_derive(StoreUtils, attributes(default_options_override_fn))]
+pub fn derive_store_utils(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as ItemStruct);
+    let name = &input.ident;
+    let generics = &input.generics;
+    let generics_names = extract_generics_names(generics);
+
+    let (field_names, inner_types, derived_table_options, simple_field_type_name) =
+        extract_struct_info(input.clone(), "Store".to_owned());
+    let default_options_override_fn_names: Vec<proc_macro2::TokenStream> = derived_table_options
+        .iter()
+        .map(|q| {
+            let GeneralTableOptions::OverrideFunction(fn_name) = q;
+            fn_name.parse().unwrap()
+        })
+        .collect();
+
+    let generics_bounds =
+        "core::hash::Hash + core::cmp::Eq + std::fmt::Debug + serde::Serialize + for<'de> serde::de::Deserialize<'de> + core::marker::Send + 'static";
+    let generics_bounds_token: proc_macro2::TokenStream = generics_bounds.parse().unwrap();
+
+    let config_struct_name_str = format!("{}Configurator", name);
+    let config_struct_name: proc_macro2::TokenStream = config_struct_name_str.parse().unwrap();
```

Comment:
By and large, every time you use `parse()`, ask yourself if you can reformulate with [parse_quote](https://docs.rs/syn/latest/syn/macro.parse_quote.html)